Python.md

```diff
@@ -15,23 +20,63 @@ As most of the other projects written in Python, we use [PEP 8]
 (https://www.youtube.com/watch?v=wf-BqAjZb8M)" by Raymond Hettinger at PyCon 2015,
 which talks about the famous P versus NP problem.*
 
+If the project has a `pylintrc` file, you can use pylint to check for potential style violations:
+
+    pylint --rcfile=pylintrc
+
 The general idea is to make the code within a project consistent and easy to interpret (for humans).
 
+
 ### Docstrings
 
-Unless the function is easy to understand quickly, it should have a docstring
-describing what it does, how it does it, what the arguments are, and what the
-expected output is.
+Include docstrings in functions. This should say *what* the function does, its *inputs*,
+and *outputs* or possible *failure states* (exceptions).
 
-Most of our Python projects use "Google-style" docstrings:
+Unless otherwise stated, we use "Google-style" docstrings:
 https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments.
 
+For flask application docstrings for views are turned into API Documentation
+(See an example for AcousticBrainz at http://acousticbrainz.readthedocs.io/)
+
+View docstrings should be formatted to include query format and an example of the response:
+
+```python
+"""Get low-level data for many recordings at once.
+
+**Example response**:
+
+.. sourcecode:: json
+
+   {"mbid1": {"some": "json"}}
+
+:query recording_ids: *Required.* A list of recording MBIDs to retrieve
+
+  Takes the form `mbid[:offset];mbid[:offset]`. Offsets are optional, and should
+  be >= 0
+:resheader Content-Type: *application/json*
+"""
+```
+
+An example of a docstring for a regular method
+```python
+def frob(foo, bar=None):
+    """Frob a foo, with an optional bar operation
+       Args:
+           foo: the foo to be frobbed
+           bar: if set, a callback to a bar operation to be applied after
+                the foo is frobbed
+       Returns:
+           The new foo, after frobbing and baring
```

Comment:
And proper capitalisation with punctuation. After all, it's for humans and should be easy to read.